### PR TITLE
Handle selection for multiple values in tree

### PR DESCRIPTION
**Pull request by aaomidi**
Potentially a fix for #5388 

However, this is still a good fix to have if it doesn't fix that. Before, if you had two servers in the list and single clicked each one within 300 ms. The code treated it like you double clicked the second item in the list. This makes it so the selections are handled per item, rather than globally.

### CODE REVIEW

**Comment by aaomidi** [comment]
Hold off on reviewing, going to write this in a cleaner way.

File: src/sql/workbench/parts/objectExplorer/browser/treeSelectionHandler.ts [hunk 1]
```diff
@@ -42,29 +42,42 @@ export class TreeSelectionHandler {
...
-				let isDoubleClick = this._clicks > 1;
+				let isDoubleClick = this._clicks.get(specificSelection) > 1;
 				this.handleTreeItemSelected(connectionManagementService, objectExplorerService, isDoubleClick, isKeyboard, selection, tree, connectionCompleteCallback);
 			}
-			this._clicks = 0;
-			this._doubleClickTimeoutTimer = undefined;
-		}, 300);
+			this._clicks.delete(specificSelection);
+
+			process.nextTick(() => {
```
Comment (anthonydresser) [inline src/sql/workbench/parts/objectExplorer/browser/treeSelectionHandler.ts]: Don't use process here, user setImmediate

Comment (aaomidi) [reply src/sql/workbench/parts/objectExplorer/browser/treeSelectionHandler.ts]: Changed how the code works entirely

**Comment by aaomidi** [comment]
Double click events are emitted differently in windows. I wonder if we can pass through those events instead.

**Pull request merged**